src/platform/x86_64/x86_64.zig

```diff
@@ -41,8 +41,36 @@ pub fn platform_init() !void {
 pub fn platform_early_init() void {
   serial.init();
   try interrupts.init_interrupts();
+  os.platform.smp.prepare();
+  os.thread.scheduler.init(&bsp_task);
   setup_gdt();
   os.memory.paging.init();
+
+  asm volatile("sti");
+}
+
+pub fn ap_init() void {
+  os.memory.paging.CurrentContext.apply();
+  try interrupts.init_interrupts();
+  setup_gdt();
+
+  asm volatile("sti");
+}
+
+var bsp_task: os.thread.Task = .{};
+
+pub fn self_exited() !?*os.thread.Task {
```

Comment:
Why error here?

Reply:
For flexibility for the implementation, but I guess that should be a `@panic()` instead